### PR TITLE
feat: Add support for table_format in table properties

### PR DESCRIPTION
**Pull request by robert8138**
In order to support Spark DDL with Iceberg, we need to add a new property for `USING ICEBERG` for the table parsing to succeed. This PR adds that new property. Feedback is welcome.

```python
In [1]: import sqlglot

In [2]: 
   ...: str = """
   ...: CREATE TABLE IF NOT EXISTS icehouse.iceberg_dev.wap_test_20220609 (
   ...:     id BIGINT COMMENT 'main id key',
   ...:     ds STRING COMMENT 'date'
   ...: )
   ...: USING ICEBERG
   ...: PARTITIONED BY (ds)
   ...: STORED AS PARQUET
   ...: TBLPROPERTIES (
   ...:     'write.wap.enabled'='true',
   ...:     'abb_retention_config_json'='{ "policy" : "delete_by_ds", "days": 10 }'
   ...: );
   ...: """

In [3]: sqlglot.parse(str)
Out[3]: 
[(CREATE this: 
   (SCHEMA this: 
     (TABLE this: 
       (IDENTIFIER this: wap_test_20220609, quoted: False), db: 
       (IDENTIFIER this: iceberg_dev, quoted: False), catalog: 
       (IDENTIFIER this: icehouse, quoted: False)), expressions: 
     (COLUMNDEF this: 
       (IDENTIFIER this: id, quoted: False), kind: 
       (DATATYPE this: Type.BIGINT, nested: False), comment: 
       (LITERAL this: main id key, is_string: True)), 
     (COLUMNDEF this: 
       (IDENTIFIER this: ds, quoted: False), kind: 
       (DATATYPE this: Type.TEXT, nested: False), comment: 
       (LITERAL this: date, is_string: True))), kind: TABLE, exists: True, properties: 
   (PROPERTIES expressions: 
     (PROPERTY this: 
       (LITERAL this: TABLE_FORMAT, is_string: True), value: 
       (LITERAL this: ICEBERG, is_string: True)), 
     (PROPERTY this: 
       (LITERAL this: PARTITIONED_BY, is_string: True), value: 
       (SCHEMA expressions: 
         (IDENTIFIER this: ds, quoted: False))), 
     (PROPERTY this: 
       (LITERAL this: FORMAT, is_string: True), value: 
       (LITERAL this: PARQUET, is_string: True)), 
     (PROPERTY this: 
       (LITERAL this: write.wap.enabled, is_string: True), value: 
       (LITERAL this: true, is_string: True)), 
     (PROPERTY this: 
       (LITERAL this: abb_retention_config_json, is_string: True), value: 
       (LITERAL this: { "policy" : "delete_by_ds", "days": 10 }, is_string: True))))]

In [8]: print(sqlglot.parse(str)[0].sql(dialect='spark', pretty=True, identify=True))
CREATE TABLE IF NOT EXISTS `icehouse`.`iceberg_dev`.`wap_test_20220609` (
  `id` LONG COMMENT 'main id key',
  `ds` STRING COMMENT 'date'
)
USING ICEBERG
PARTITIONED BY (
  `ds`
)
STORED AS PARQUET
TBLPROPERTIES (
  'write.wap.enabled' = 'true',
  'abb_retention_config_json' = '{ "policy" : "delete_by_ds", "days": 10 }'
)
```

### CODE REVIEW

File: sqlglot/dialects/hive.py [hunk 1]
```diff
@@ -54,11 +54,14 @@ def _properties_sql(self, expression):
     expression = expression.copy()
     properties = expression.expressions
 
+    using = ""
     stored_as = ""
     partitioned_by = ""
 
     for p in properties:
-        if p.text("this").upper() == c.FORMAT:
+        if p.text("this").upper() == c.TABLE_FORMAT:
```
Comment (tobymao) [inline sqlglot/dialects/hive.py:62]: change this to p.name

File: sqlglot/dialects/hive.py [hunk 2]
```diff
@@ -54,11 +54,14 @@ def _properties_sql(self, expression):
...
+            using = p
+        if p.text("this").upper() == c.FILE_FORMAT:
```
Comment (tobymao) [inline sqlglot/dialects/hive.py:64]: elif p.name

File: sqlglot/parser.py [hunk 3]
```diff
@@ -560,6 +560,15 @@ def _parse_properties(self, schema):
             properties.extend(self._parse_csv(lambda: self._parse_property(schema)))
             self._match_r_paren()
         else:
+            if self._match(TokenType.USING):
+                properties.append(
+                    self.expression(
+                        exp.Property,
+                        this=exp.Literal.string(c.TABLE_FORMAT),
+                        value=exp.Literal.string(self._parse_var().text("this")),
```
Comment (tobymao) [inline sqlglot/parser.py:568]: self._parse_var().name

File: tests/fixtures/identity.sql [hunk 4]
```diff
@@ -381,6 +381,8 @@ CREATE TABLE z (a INT) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARACTER SET=utf8
 CREATE TABLE z (a INT DEFAULT NULL, PRIMARY KEY(a)) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARACTER SET=utf8 COLLATE=utf8_bin COMMENT='x'
 CREATE TABLE z WITH (format = 'parquet') AS SELECT 1
 CREATE TABLE z WITH (format = 'ORC', x = '2') AS SELECT 1
+CREATE TABLE z WITH (table_format = 'iceberg', format = 'parquet') AS SELECT 1
+CREATE TABLE z WITH (table_format = 'iceberg', format = 'ORC', x = '2') AS SELECT 1
```
Comment (tobymao) [inline tests/fixtures/identity.sql:385]: add a test to dialects to test the other way around

**Comment by tobymao** [comment]
looks good, just needs some more tests

File: tests/fixtures/identity.sql (hunk 4 above)
Comment (robert8138) [reply tests/fixtures/identity.sql:385]: good call

**Review by tobymao — APPROVED** [review]

**Pull request merged**